### ISSUE TITLE
Another~Another only missing credentials problem:D

### ISSUE THREAD

**Issue by JonathanSum** [jaredhanson/passport-local#163]
I follow the tutorial from here and this is the code:
https://www.youtube.com/watch?v=mAOxWf36YLo

const express = require('express')
const router = express.Router();
const bcrypt = require('bcryptjs')
const passport = require('passport')
// const bodyParser = require('body-parser')
// Bring in User Models
let User = require('../models/user')
// router.use(express.json());
// router.use(express.urlencoded());
// router.use(bodyParser.urlencoded({
//   extended: false
// }))
// parse application/json
// router.use(bodyParser.json())

// Register form
router.get('/register', (req, res) => {
  res.render('register');
});
//Login Redirect Page
// router.post('/register', function(req, res) {
//
//     }

//REGISTER Proccess
router.post('/register', function(req, res) {
  const name = req.body.name;
  const email = req.body.email;
  const username = req.body.username;
  const password = req.body.password;
  const password2 = req.body.password2;

  req.checkBody('name', 'Name is required').notEmpty();
  req.checkBody('email', 'Email is required').notEmpty();
  req.checkBody('email', 'Email is not valid').isEmail();
  req.checkBody('username', 'Username is required').notEmpty();
  req.checkBody('password', 'Password is required').notEmpty();
  req.checkBody('password2', 'Password do not match').equals(req.body.password);
  let errors = req.validationErrors();

  if (errors) {
    res.render('register', {
      errors: errors
    })
  } else {
    let newUser = new User({
      name: name,
      email: email,
      username: username,
      password: password
    })

    bcrypt.genSalt(10, function(err, salt) {
      bcrypt.hash(newUser.password, salt, function(err, hash) {
        if (err) {
          console.log(err);
        }
        newUser.password = hash;
        newUser.save(function(err) {
          if (err) {
            console.log(err);
            return;
          } else {
            req.flash('success', 'You are Now registered and can log in');
            res.redirect('/users/login');
          }
        })
      });
    })
  }
})

//Form of Login
router.get('/login', function(req, res) {
  res.render('login');
})

// Login Proccess
router.post('/login', function(req, res, next) {
  passport.authenticate('local', {
    sucessRedirect: '/',
    failureRedirect: '/users/login',
    failureFlash: true
  })(req, res, next)
})

module.exports = router;

**Comment by JonathanSum** [comment]
NVM, I found the problem and fixed it. Sorry, everyone and owner.

**Comment by JonathanSum** [comment]
I found the problem to find it. Thus, forget about this. But, if someone can, plz explain why we put the secret in the MongoDB.

**Comment by MauriceMorrey** [comment]
Were you ever able to fix this?

**Comment by Croma1994** [comment]
Another one banging the head against the wall...

**Comment by JonathanSum** [comment]
Good Question! @MauriceMorrey  @Croma1994 
I remember I fix it. But if you look at the date, do you still remember how to fix it if it is you? But don't worry! The solution must be in my GitHub repo because I usually upload almost everything I do. So the solution so far is to find the same piece of code in my Github, and you will see a fixed solution. 


So far I don't want to spend too much time finding it on my repo, but if we have too many people have this issue, I may find it, test it, and debug it. 

So far I don't want to spend too much time on Web Dev but in Deep Learning.
![Satou_492 copy](https://user-images.githubusercontent.com/21982975/85205499-860a8700-b2d0-11ea-8ef5-698c9ebcddcb.png)

**Comment by Croma1994** [comment]
So far, in my case was just passing this into the local strategy...

```
    {
      usernameField: "name",
      passwordField: "password",
      passReqToCallback: true,
    }
```

**Comment by JonathanSum** [comment]
@Croma1994 does it work?  If it works, it is great.

**Comment by Croma1994** [comment]
@JonathanSum Yep! I had a misconception, thought that the local strategy function searched automatically for the req.body payloads no matter what name attribute they had, but you have to pass that object specifying the name and passing the req to the callback in order to use flash messages.

**Comment by sourav-besra-vst-au4** [comment]
Didn't work for me, i have been banging my head for 3 days?
Any fix guys?

**Comment by JonathanSum** [comment]
Yeah. Another one has this issue. Welcome on the board.
![image](https://user-images.githubusercontent.com/21982975/90271235-6aaa8c80-dde6-11ea-956d-e7a5ce4f77ad.png)

**Comment by jefflaub12** [comment]
Also same issue

passport.use('local.signup', new LocalStrategy({
    usernameField: 'email',
    passwordField: 'password',
    passReqToCallback: true
}, function(req, email, password, done) {
    req.check('email', 'Invalid Email').notEmpty().isEmail();
    req.check('password', 'Your password must be at least 5 characters').notEmpty().isLength({min:5});

**Comment by MauriceMorrey** [comment]
I don't quite remember the fix but it had something to do with passing parameters named differently; in the documentation link @jefflaub12 shared, it says;

"By default, LocalStrategy expects to find credentials in parameters named username and password. If your site prefers to name these fields differently, options are available to change the defaults."

I know it says you have the option to change defaults but if I remember correctly, using the defaults fixed the issue for me.

Good luck.

**Comment by jefflaub12** [comment]
That did it used the defaults instead of attempting to rename the variable for usernameField. Thanks
